subprojects/kotlin-dsl-integ-tests/kotlin-dsl-integ-tests.gradle.kts

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import org.gradle.gradlebuild.test.integrationtests.IntegrationTest
+import org.gradle.gradlebuild.unittestandcompile.ModuleType
+import plugins.futurePluginVersionsFile
+
+plugins {
+    `kotlin-library`
+}
+
+description = "Kotlin DSL Integration Tests"
+
+gradlebuildJava {
+    moduleType = ModuleType.INTERNAL
+}
+
+dependencies {
+
+    testImplementation(project(":kotlinDslTestFixtures"))
+
+    integTestImplementation("com.squareup.okhttp3:mockwebserver:3.9.1")
+
+    val allTestRuntimeDependencies: DependencySet by rootProject.extra
```

Comment:
:o: This code is replicated in more than 3 places now. I don't think we need to fix it before merging but we should clean it up afterwards with something like `forAllTestRuntimeDependencies { integTestRuntimeOnly(it) }`.

Reply:
Yep, agreed it's not ideal. Same thing for both `allTestRuntimeDependencies` and `allCoreRuntimeExtensions`. A better pattern to manage this would be nice. Adding a function in `buildSrc` feels weird because the declarations comes from the root build script. Or it would have to take the extra property name as a parameter, not making the whole thing much better. I'd postpone changing this.

Reply:
I consider ok for the root script to depend on conventions defined in `buildSrc`, wouldn't that be similar to how we did it in the `kotlin-dsl` build with `build.loadExtraPropertiesOf`, etc?

Agreed to postpone 👍